### PR TITLE
Add jobs for Cluster API v1.5

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add jobs for Cluster API v1.5. More information on this release can be found at the below links.

- https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.0-beta.0
- https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/releases/release-1.5.md

### CODE REVIEW

**Comment by killianmuldoon** [comment]
/hold 

Still working on this one.

**Comment by killianmuldoon** [comment]
retest

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml [hunk 1]
```diff
@@ -0,0 +1,202 @@
+periodics:
+- name: periodic-cluster-api-test-release-1-5
+  interval: 2h
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml:3]: Q: Should the interval be 4h (like for release-1.4?)

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5-upgrades.yaml [hunk 2]
```diff
@@ -0,0 +1,319 @@
+periodics:
+
+- name: periodic-cluster-api-e2e-workload-upgrade-1-22-1-23-1-5
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5-upgrades.yaml:3]:
```suggestion
- name: periodic-cluster-api-e2e-workload-upgrade-1-22-1-23-release-1-5
```
similar below

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Comment by fabriziopandini** [comment]
Great work!
/lgtm
/approve

**Pull request merged**